### ISSUE TITLE
type assertion error could occur

### ISSUE THREAD

**Issue by takuhiro** [NetApp/trident#802]
**Describe the bug**

We hit the following error in our environment just once. 
```
time="2023-02-01T02:04:48Z" level=fatal msg="could not perform assertion: hybridControllerFrontend.(controllerhelpers.ControllerHelper)"
```
As I can see the trident log, the process exited by the fatal error and then the trident container was restarted successfully.

[`hybridControllerFrontend`](https://github.com/NetApp/trident/blob/v23.01.0/main.go#L366) could be nil if a temporary error(e.g. failure of creating k8s client) occurs [here](https://github.com/NetApp/trident/blob/v23.01.0/main.go#L352), which causes the type assertion error.
The `err` is checked [here](https://github.com/NetApp/trident/blob/v23.01.0/main.go#L358), but before that, the `err` is overwitten [here](https://github.com/NetApp/trident/blob/v23.01.0/main.go#L353). So if `k8sctrlhelper.NewHelper` failed but `k8snodehelper.NewHelper` succeeded, the error check would be passed and the type assertion of `hybridControllerFrontend` would be executed even though it is nil.

**Environment**
Provide accurate information about the environment to help us reproduce the issue.

- Trident version: v23.01.0
- Trident installation flags used: `silenceAutosupport: true` (Trident Operator)
- Container runtime: containerd 1.6.10
- Kubernetes version: v1.25.6
- Kubernetes orchestrator: none
- Kubernetes enabled feature gates: none
- OS: Ubuntu 20.04.5 LTS
- NetApp backend types: ONTAP 9.9.1P9
- Other:

**To Reproduce**
It is difficult to reproduce this problem intentionally, but I think it would be reproduced if you changed the code to make k8sctrlhelper.NewHelper always return error [here](https://github.com/NetApp/trident/blob/v23.01.0/frontend/csi/controller_helpers/kubernetes/plugin.go#L113-L115).

**Expected behavior**
handle the error correctly

**Additional context**

**Comment by gnarl** [comment]
Hi @takuhiro,

This issue is fixed with [commit afb725a](https://github.com/NetApp/trident/commit/afb725af4f34a610d976484cf08611c4ca7a38fd) and will be included in the Trident v23.04 release.

**Comment by takuhiro** [comment]
Thank you @gnarl !
